internal/cli/atlas/backup/schedule/update.go

```diff
@@ -43,35 +56,163 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
 	}
 }
 
-func (opts *UpdateOpts) Run() error {
-	r, err := opts.store.UpdateSchedule(opts.ConfigProjectID(), opts.clusterName, &atlas.CloudProviderSnapshotBackupPolicy{})
+func (opts *UpdateOpts) Run(cmd *cobra.Command) error {
+	backupConfig, err := opts.NewBackupConfig(cmd, opts.clusterName)
+	if err != nil {
+		return err
+	}
+
+	r, err := opts.store.UpdateSchedule(opts.ConfigProjectID(), opts.clusterName, backupConfig)
 	if err != nil {
 		return err
 	}
 
 	return opts.Print(r)
 }
 
+func (opts *UpdateOpts) NewBackupConfig(cmd *cobra.Command, clusterName string) (*atlas.CloudProviderSnapshotBackupPolicy, error) {
+	out := new(atlas.CloudProviderSnapshotBackupPolicy)
+
+	out.ClusterName = clusterName
+	opts.verifyExportBucketID(out)
+
+	if cmd.Flags().Changed(flag.ExportFrequencyType) {
+		checkForExport(out)
+		out.Export.FrequencyType = opts.exportFrequencyType
+	}
+	if cmd.Flags().Changed(flag.ReferenceHourOfDay) {
+		out.ReferenceHourOfDay = &opts.referenceHourOfDay
+	}
+	if cmd.Flags().Changed(flag.ReferenceMinuteOfHour) {
+		out.ReferenceMinuteOfHour = &opts.referenceMinuteOfHour
+	}
+	if cmd.Flags().Changed(flag.RestoreWindowDays) {
+		out.RestoreWindowDays = &opts.restoreWindowDays
+	}
+
+	out.AutoExportEnabled = returnValueForSetting(opts.autoExport, opts.noAutoExport)
+	out.UpdateSnapshots = returnValueForSetting(opts.updateSnapshots, opts.noUpdateSnapshots)
+	out.UseOrgAndGroupNamesInExportPrefix = returnValueForSetting(opts.useOrgAndGroupNamesInExportPrefix, opts.noUseOrgAndGroupNamesInExportPrefix)
+
+	return out, nil
+}
+
+func (opts *UpdateOpts) verifyExportBucketID(out *atlas.CloudProviderSnapshotBackupPolicy) {
```

Comment:
seems that we use verify for validations and now verify for assigning, how about moving this code to `NewBackupConfig` ?

Reply:
Good idea. Name of this function is misleading.